File: Melo/Melo.Models/Models/Genre/GenreResponse.cs
namespace Melo.Models
{
	public class GenreResponse
	{ 
		public int Id { get; set; }

		public DateTime? CreatedAt { get; set; }

		public string? CreatedBy { get; set; }

		public DateTime? ModifiedAt { get; set; }

		public string? ModifiedBy { get; set; }

		public string? Name { get; set; }

		public long? ViewCount { get; set; }

		public string? ImageUrl { get; set; }
	}
}

File: Melo/Melo.UI/melo_desktop/lib/models/genre_response.dart
class GenreResponse {
  final int id;
  final String? name;
  final int? viewCount;
  final String? imageUrl;

  GenreResponse({
    required this.id,
    this.name,
    this.viewCount,
    this.imageUrl,
  });

  factory GenreResponse.fromJson(Map<String, dynamic> json) {
    return GenreResponse(
      id: json['id'],
      name: json['name'],
      viewCount: json['viewCount'],
      imageUrl: json['imageUrl'],
    );
  }
}
